src/Microsoft.IdentityModel.Tokens/AppContextSwitches.cs

```diff
@@ -16,6 +16,54 @@ internal static class AppContextSwitches
         /// </summary>
         internal const string UseClaimsIdentityTypeSwitch = "Microsoft.IdentityModel.Tokens.UseClaimsIdentityType";
 
-        internal static bool UseClaimsIdentityType() => (AppContext.TryGetSwitch(UseClaimsIdentityTypeSwitch, out bool useClaimsIdentityType) && useClaimsIdentityType);
+        private static bool? _useClaimsIdentity;
+
+        internal static bool UseClaimsIdentityType => _useClaimsIdentity ??= (AppContext.TryGetSwitch(UseClaimsIdentityTypeSwitch, out bool useClaimsIdentityType) && useClaimsIdentityType);
+
+        /// <summary>
+        /// Used for testing to reset the <see cref="UseClaimsIdentityType"/> switch to its default value.
+        /// </summary>
+        internal static void ResetUseClaimsIdentityTypeSwitch()
+        {
+            _useClaimsIdentity = null;
+            AppContext.SetSwitch(UseClaimsIdentityTypeSwitch, false);
+        }
+
+        /// <summary>
+        /// When validating the issuer signing key, specifies whether to fail if the 'tid' claim is missing.
+        /// </summary>
+        internal const string DontFailOnMissingTidSwitch = "Switch.Microsoft.IdentityModel.DontFailOnMissingTidValidateIssuerSigning";
```

Comment:
thought Brent had a comment to use "DoNot" instead of "Dont" somewhere?

Reply:
Yep, done, variable names changed. The string is left the same, though.